### PR TITLE
[SLAM] Enable on error event

### PR DESCRIPTION
**Pull request by lisa0314**
@DonnaWuDongxia, please help to review.

### CODE REVIEW

**Comment by kenny-y** [comment]
Can this be merged?

**Comment by DonnaWuDongxia** [comment]
Generally, looks good to me on the implementation. Please remember to update the doc and fix the related bug if needed.

File: src/slam/slam_module_dev.cpp [hunk 1]
```diff
@@ -233,7 +248,13 @@ utils::Status SlamModuleDev::SetMotionCallbacks() {
...
 
         if (mw->process_sample_set(sampleSet) < rs::core::status_no_error) {
-          utils::OnSlamError("failed to process sample");
+          auto payload = new ErrorEventPayload(
+            SlamRunnerDev::GetSlamRunner(),
+            new std::string("failed to process sample"));
+          AsyncTaskRunnerInstance::GetInstance()->PostTask(
+            new ErrorEventTask(),
+            payload,
+            "{{EEROR_EVENT MESSAGE}}");
```
Comment (DonnaWuDongxia) [inline src/slam/slam_module_dev.cpp]: If we use these lines so many times, maybe we can make it a member function.

**Comment by lisa0314** [comment]
@DonnaWuDongxia , thanks for your review. Would you like to help to merge it?

**Comment by DonnaWuDongxia** [comment]
@lisa0314  Please remember to update the doc and fix the related bug if needed in this single PR.

File: src/slam/slam_module_dev.cpp [hunk 2]
```diff
@@ -161,7 +172,10 @@ utils::Status SlamModuleDev::SetStreamCallbacks() {
             + std::to_string(static_cast<int>(stream))
             + std::string("\twith frame counter:")
             + std::to_string(frame.get_frame_number());
-        utils::OnSlamError(utils::Status(SLAM_ADDON_ERROR, errorMessage));
+        auto payload = new ErrorEventPayload(
+            SlamRunnerDev::GetSlamRunner(),
+            new std::string(errorMessage));
+        PostErrorEventTask(payload);
```
Comment (DonnaWuDongxia) [inline src/slam/slam_module_dev.cpp]: How about just take a string as the parameter for `PostErrorEventTask` method?

**Comment by DonnaWuDongxia** [comment]
Looks good to me! @kenny-y @halton I think it's ready to merge.

**Comment by halton** [comment]
@DonnaWuDongxia please let me know when this PR has your LGTM

**Comment by DonnaWuDongxia** [comment]
It's ready. @halton

**Comment by halton** [comment]
Next time, please give explicit LGTM

**Pull request merged**